checkers/testdata/sortSlice/negative_tests.go

```diff
@@ -0,0 +1,106 @@
+package checker_test
+
+import (
+	"sort"
+)
+
+func goodSorting() {
+	{
```

Comment:
I think we should not support `f(x)`, isn't it?
```
var xs []int
sort.Slice(ys, func(i, j int) bool {
    return foo(ys[i]) >= foo(ys[j])
})
```

Reply:
This case should not give any warnings in the current implementation due to several checks:
1. return expression contains side effects (function calls).
2. there are `ys` inside both side of the `>=` operator.

We can't give a warning for that case since there are valid cases that involve key generation based on the element.

Imagine that we want to sort ints using string-like order:
```
var xs []int
sort.Slice(ys, func(i, j int) bool {
    return int2string(ys[i]) >= int2string(ys[j])
})
```
The only case we're trying to catch is using conditions that do not depend on the slice being sorted.

Reply:
My idea was to add such test (skipping func calls), unless I've missed it.

Reply:
There is this:
![image](https://user-images.githubusercontent.com/6286655/82938022-d294c800-9f99-11ea-9d08-73d848fc9710.png)

But I can add another one with slices in function calls.

Reply:
Done.